--- configure.orig	2012-08-12 02:21:17.764958843 +0200
+++ configure	2012-08-12 02:21:40.400944475 +0200
@@ -10534,7 +10534,7 @@
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for ICONV support" >&5
 $as_echo_n "checking for ICONV support... " >&6; }
 
-	if test -f /usr/include/iconv.h; then
+	if test -f /usr/include/sys/iconv.h; then
 		found_iconv="yes"
 	elif test -f /usr/local/include/iconv.h; then
 		ICONV_CFLAGS="-I/usr/local/include"
