--- ssh.c.orig	2012-04-16 13:53:24.000000000 +0200
+++ ssh.c	2012-04-18 10:39:14.000000000 +0200
@@ -267,14 +267,25 @@
 			fatal("setrlimit failed: %.100s", strerror(errno));
 	}
 #endif
-	/* Get user data. */
-	pw = getpwuid(original_real_uid);
-	if (!pw) {
-		logit("You don't exist, go away!");
-		exit(255);
-	}
-	/* Take a copy of the returned structure. */
-	pw = pwcopy(pw);
+
+    /* Create record for passwd structure on our own.. We don't have passwd support in ServeD */
+    pw = (struct passwd *)malloc(sizeof(struct passwd));
+    if (pw == 0) {
+        fatal("Unable to allocate memory for passwd struct.");
+    }
+    char homeDir[64];
+    char userName[64];
+    sprintf(homeDir, "%s", getenv("HOME"));
+    sprintf(userName, "%s", getenv("USER"));
+
+
+    pw->pw_uid = original_real_uid;
+    pw->pw_gid = 0; /* wheel */
+    pw->pw_name = (char *)malloc((strlen(userName)+1)*sizeof(char));
+    sprintf(pw->pw_name, "%s", userName);
+    pw->pw_dir = (char *)malloc((strlen(homeDir)+1)*sizeof(char));
+    sprintf(pw->pw_dir, "%s", homeDir);
+

 	/*
 	 * Set our umask to something reasonable, as some files are created
@@ -1481,8 +1492,23 @@
 		xfree(keys);
 	}
 #endif /* ENABLE_PKCS11 */
-	if ((pw = getpwuid(original_real_uid)) == NULL)
-		fatal("load_public_identity_files: getpwuid failed");
+
+    pw = (struct passwd *)malloc(sizeof(struct passwd));
+    if (pw == 0) {
+        fatal("Unable to allocate memory for passwd struct.");
+    }
+    char homeDir[64];
+    char userName[64];
+    sprintf(homeDir, "%s", getenv("HOME"));
+    sprintf(userName, "%s", getenv("USER"));
+
+    pw->pw_uid = original_real_uid;
+    pw->pw_gid = 0; /* wheel */
+    pw->pw_name = (char *)malloc((strlen(userName)+1)*sizeof(char));
+    sprintf(pw->pw_name, "%s", userName);
+    pw->pw_dir = (char *)malloc((strlen(homeDir)+1)*sizeof(char));
+    sprintf(pw->pw_dir, "%s", homeDir);
+
 	pwname = xstrdup(pw->pw_name);
 	pwdir = xstrdup(pw->pw_dir);
 	if (gethostname(thishost, sizeof(thishost)) == -1)
