--- scp.c.orig	2012-04-25 19:49:10.000000000 +0200
+++ scp.c	2012-04-25 19:57:39.000000000 +0200
@@ -477,8 +477,23 @@
 	argc -= optind;
 	argv += optind;

-	if ((pwd = getpwuid(userid = getuid())) == NULL)
-		fatal("unknown user %u", (u_int) userid);
+    /* Create record for passwd structure on our own.. We don't have passwd support in ServeD */
+    int uid = getuid();
+    pwd = (struct passwd *)malloc(sizeof(struct passwd));
+    if (pwd == 0) {
+        fatal("Unable to allocate memory for passwd struct.");
+    }
+    char homeDir[64];
+    char userName[64];
+    sprintf(userName, "%s", getenv("USER"));
+    sprintf(homeDir, "%s", getenv("HOME"));
+
+    pwd->pw_uid = uid;
+    pwd->pw_gid = 0; /* wheel */
+    pwd->pw_name = (char *)malloc((strlen(userName)+1)*sizeof(char));
+    sprintf(pwd->pw_name, "%s", userName);
+    pwd->pw_dir = (char *)malloc((strlen(homeDir)+1)*sizeof(char));
+    sprintf(pwd->pw_dir, "%s", homeDir);

 	if (!isatty(STDOUT_FILENO))
 		showprogress = 0;
