--- ident_orig.c	2012-03-11 00:31:47.000000000 +0100
+++ ident.c	2012-03-11 00:34:32.000000000 +0100
@@ -124,8 +124,8 @@
 	/* Get the name ("gecos") */
 	if (!*name && !git_default_name[0]) {
 		pw = getpwuid(getuid());
-		if (!pw)
-			die("You don't exist. Go away!");
+		if (pw)
+            // die("You don't exist. Go away!");
 		copy_gecos(pw, git_default_name, sizeof(git_default_name));
 	}
 	if (!*name)
@@ -141,8 +141,8 @@
 		} else {
 			if (!pw)
 				pw = getpwuid(getuid());
-			if (!pw)
-				die("You don't exist. Go away!");
+			if (pw)
+                // die("You don't exist. Go away!");
 			copy_email(pw);
 		}
 	}
@@ -256,11 +256,12 @@
 		if (error_on_no_name)
 			die("empty ident %s <%s> not allowed", name, email);
 		pw = getpwuid(getuid());
-		if (!pw)
-			die("You don't exist. Go away!");
-		strlcpy(git_default_name, pw->pw_name,
-			sizeof(git_default_name));
-		name = git_default_name;
+		if (pw) {
+            // die("You don't exist. Go away!");
+    		strlcpy(git_default_name, pw->pw_name,
+    			sizeof(git_default_name));
+    		name = git_default_name;    
+        }
 	}
 
 	strcpy(date, git_default_date);
