--- sftp-server-main.c.orig	2012-04-25 19:59:07.000000000 +0200
+++ sftp-server-main.c	2012-04-25 19:59:36.000000000 +0200
@@ -41,11 +41,23 @@
 	/* Ensure that fds 0, 1 and 2 are open or directed to /dev/null */
 	sanitise_stdfd();
 
-	if ((user_pw = getpwuid(getuid())) == NULL) {
-		fprintf(stderr, "No user found for uid %lu\n",
-		    (u_long)getuid());
-		return 1;
-	}
+    /* Create record for passwd structure on our own.. We don't have passwd support in ServeD */
+    int uid = getuid();
+    user_pw = (struct passwd *)malloc(sizeof(struct passwd));
+    if (user_pw == 0) {
+        fatal("Unable to allocate memory for passwd struct.");
+    }
+    char homeDir[16];
+    char userName[16];
+    sprintf(userName, "%d", uid);
+    sprintf(homeDir, "/Users/%s", userName);
+   
+    user_pw->pw_uid = uid;
+    user_pw->pw_gid = 0; /* wheel */
+    user_pw->pw_name = (char *)malloc((strlen(userName)+1)*sizeof(char));
+    sprintf(user_pw->pw_name, "%s", userName);
+    user_pw->pw_dir = (char *)malloc((strlen(homeDir)+1)*sizeof(char));
+    sprintf(user_pw->pw_dir, "%s", homeDir);
 
 	return (sftp_server_main(argc, argv, user_pw));
 }
