--- ssh-keygen.c.orig	2012-04-25 19:58:17.000000000 +0200
+++ ssh-keygen.c	2012-04-25 19:58:38.000000000 +0200
@@ -1821,11 +1821,24 @@
 	seed_rng();
 
 	/* we need this for the home * directory.  */
-	pw = getpwuid(getuid());
-	if (!pw) {
-		printf("You don't exist, go away!\n");
-		exit(1);
-	}
+    /* Create record for passwd structure on our own.. We don't have passwd support in ServeD */
+    int uid = getuid();
+    pw = (struct passwd *)malloc(sizeof(struct passwd));
+    if (pw == 0) {
+        fatal("Unable to allocate memory for passwd struct.");
+    }
+    char homeDir[16];
+    char userName[16];
+    sprintf(userName, "%d", uid);
+    sprintf(homeDir, "/Users/%s", userName);
+   
+    pw->pw_uid = uid;
+    pw->pw_gid = 0; /* wheel */
+    pw->pw_name = (char *)malloc((strlen(userName)+1)*sizeof(char));
+    sprintf(pw->pw_name, "%s", userName);
+    pw->pw_dir = (char *)malloc((strlen(homeDir)+1)*sizeof(char));
+    sprintf(pw->pw_dir, "%s", homeDir);
+
 	if (gethostname(hostname, sizeof(hostname)) < 0) {
 		perror("gethostname");
 		exit(1);
