--- sapi/fpm/fpm/fpm_sockets.c.orig	2012-09-09 00:53:29.097604783 +0200
+++ sapi/fpm/fpm/fpm_sockets.c	2012-09-09 00:52:40.060573094 +0200
@@ -401,16 +401,16 @@
 	}
 
 	/* kernel >= 2.6.24 return non-zero here, that means operation is supported */
-	if (info.tcpi_sacked == 0) {
+	if (info.__tcpi_sacked == 0) {
 		return -1;
 	}
 
 	if (cur_lq) {
-		*cur_lq = info.tcpi_unacked;
+		*cur_lq = info.__tcpi_unacked;
 	}
 
 	if (max_lq) {
-		*max_lq = info.tcpi_sacked;
+		*max_lq = info.__tcpi_sacked;
 	}
 
 	return 0;
