--- src/bin/initdb/initdb.c	2011-12-01 22:49:59.000000000 +0100
+++ src/bin/initdb/initdb_patched.c	2012-02-18 16:03:01.000000000 +0100
@@ -644,9 +644,6 @@
 static char *
 get_id(void)
 {
-#ifndef WIN32
-
-	struct passwd *pw;

 	if (geteuid() == 0)			/* 0 is root's uid */
 	{
@@ -659,34 +656,15 @@
 		exit(1);
 	}

-	pw = getpwuid(geteuid());
-	if (!pw)
-	{
-		fprintf(stderr,
-			  _("%s: could not obtain information about current user: %s\n"),
-				progname, strerror(errno));
-		exit(1);
-	}
-#else							/* the windows code */
-
-	struct passwd_win32
-	{
-		int			pw_uid;
-		char		pw_name[128];
-	}			pass_win32;
-	struct passwd_win32 *pw = &pass_win32;
-	DWORD		pwname_size = sizeof(pass_win32.pw_name) - 1;
-
-	pw->pw_uid = 1;
-	if (!GetUserName(pw->pw_name, &pwname_size))
-	{
-		fprintf(stderr, _("%s: could not get current user name: %s\n"),
-				progname, strerror(errno));
-		exit(1);
-	}
-#endif
+    /**
+    *    @author dmilith
+    *
+    *   This patch is required for ServeD system. We don't use /etc/passwd at all
+    */
+    char result[64]; /* max length of name of user */

-	return xstrdup(pw->pw_name);
+    sprintf(result, "%s", getenv("USER"));
+    return xstrdup(result);
 }

 static char *
