--- src/bin/psql/command.c	2012-02-18 17:16:41.000000000 +0100
+++ src/bin/psql/command_patched.c	2012-02-18 17:15:48.000000000 +0100
@@ -258,24 +258,14 @@
 			dir = opt;
 		else
 		{
-#ifndef WIN32
-			struct passwd *pw;
-
-			pw = getpwuid(geteuid());
-			if (!pw)
-			{
-				psql_error("could not get home directory: %s\n", strerror(errno));
-				exit(EXIT_FAILURE);
-			}
-			dir = pw->pw_dir;
-#else							/* WIN32 */
-
-			/*
-			 * On Windows, 'cd' without arguments prints the current
-			 * directory, so if someone wants to code this here instead...
-			 */
-			dir = "/";
-#endif   /* WIN32 */
+        /**
+            *    @author dmilith
+            *
+            *   This patch is required for ServeD system. We don't use /etc/passwd at all
+            */
+           char result[64];
+           sprintf(result, "%s", getenv("HOME"));
+           dir = strdup(result);
 		}

 		if (chdir(dir) == -1)
