--- src/bin/scripts/common.c	2012-02-18 17:23:33.000000000 +0100
+++ src/bin/scripts/common_patched.c	2012-02-18 17:23:18.000000000 +0100
@@ -36,29 +36,15 @@
 const char *
 get_user_name(const char *progname)
 {
-#ifndef WIN32
-	struct passwd *pw;
-
-	pw = getpwuid(geteuid());
-	if (!pw)
-	{
-		fprintf(stderr, _("%s: could not obtain information about current user: %s\n"),
-				progname, strerror(errno));
-		exit(1);
-	}
-	return pw->pw_name;
-#else
-	static char username[128];	/* remains after function exit */
-	DWORD		len = sizeof(username) - 1;
+   /**
+    *    @author dmilith
+    *
+    *   This patch is required for ServeD system. We don't use /etc/passwd at all
+    */
+   char result[64]; /* max length of user name */

-	if (!GetUserName(username, &len))
-	{
-		fprintf(stderr, _("%s: could not get current user name: %s\n"),
-				progname, strerror(errno));
-		exit(1);
-	}
-	return username;
-#endif
+   sprintf(result, "%s", getenv("USER"));
+   return strdup(result);
 }


