--- src/port/thread.c	2012-02-18 17:43:02.000000000 +0100
+++ src/port/thread_patched.c	2012-02-18 17:40:26.000000000 +0100
@@ -89,27 +89,17 @@
 pqGetpwuid(uid_t uid, struct passwd * resultbuf, char *buffer,
 		   size_t buflen, struct passwd ** result)
 {
-#if defined(FRONTEND) && defined(ENABLE_THREAD_SAFETY) && defined(HAVE_GETPWUID_R)
+   /**
+    *    @author dmilith
+    *
+    *   This patch is required for ServeD system. We don't use /etc/passwd at all
+    */
+    return 0;
+}
 
-#ifdef GETPWUID_R_5ARG
-	/* POSIX version */
-	getpwuid_r(uid, resultbuf, buffer, buflen, result);
-#else
 
-	/*
-	 * Early POSIX draft of getpwuid_r() returns 'struct passwd *'.
-	 * getpwuid_r(uid, resultbuf, buffer, buflen)
-	 */
-	*result = getpwuid_r(uid, resultbuf, buffer, buflen);
-#endif
-#else
 
-	/* no getpwuid_r() available, just use getpwuid() */
-	*result = getpwuid(uid);
-#endif
 
-	return (*result == NULL) ? -1 : 0;
-}
 #endif
 
 /*
